src/compas_rhino/geometry/trimesh/slice.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import Rhino
+from Rhino.Geometry import Point3d, Vector3d, Plane
+from Rhino.Geometry.Intersect.Intersection import MeshPlane
+
+import compas
+from compas.plugins import plugin
+
+
+__all__ = [
+    'trimesh_slice',
+]
+
+
+@plugin(category="trimesh", requires=['Rhino'])
+def trimesh_slice(mesh, planes):
+    """Slice a mesh by a list of planes.
+    Parameters
+    ----------
+    mesh : tuple of vertices and faces
+        The mesh to slice.
+    planes : list of (point, normal) tuples or compas.geometry.Plane
+        The slicing planes.
+    Returns
+    -------
+    list of arrays
+        The points defining the slice polylines.
+
+    Examples
+    --------
+    >>> from compas.geometry import Sphere, Plane, Point, Vector
+    >>> from compas.datastructures import Mesh
+    >>> sphere = Sphere([1, 1, 1], 1)
+    >>> sphere = Mesh.from_shape(sphere, u=30, v=30)
+    >>> sphere.quads_to_triangles()
+    >>> sphere.to_vertices_and_faces()
```

Comment:
Since `Sphere` has a `to_vertices_and_faces` method, this can be simplified:
```suggestion
    >>> sphere = Sphere([1, 1, 1], 1)
    >>> sphere = sphere.to_vertices_and_faces(u=30, v=30)
```